src/SprykerEco/Zed/Amazonpay/Business/Api/Converter/AbstractResponseParserConverter.php

```diff
@@ -110,14 +117,15 @@ protected function extractHeader(ResponseInterface $responseParser)
         $header->setStatusCode($statusCode);
 
         if ($metadata) {
-            $header->setRequestId($metadata['RequestId']);
+            $header->setRequestId($metadata[self::REQUEST_ID]);
         }
 
-        if (!empty($responseParser->toArray()['Error'])) {
-            $responseArray = $responseParser->toArray();
-            $header->setErrorMessage($responseArray['Error']['Message']);
-            $header->setErrorCode($responseArray['Error']['Code']);
-            $header->setRequestId($responseArray['RequestId']);
+        $responseArray = $responseParser->toArray();
```

Comment:
This whole code block can go to method. 
Use `static::` whenever you using class constants.
Why only some of the response data have constants for array keys?